src/System.Text.Json/tests/JsonDocumentTests.cs

```diff
@@ -2017,6 +2561,37 @@ public static void EnsureResizeSucceeds()
             }
         }
 
+        private static void BuildSegmentedReader(
+            out Utf8JsonReader reader,
+            ReadOnlyMemory<byte> data,
+            int segmentCount,
+            in JsonReaderState state,
+            bool isFinalBlock=false)
+        {
+            if (segmentCount == 0)
+            {
+                reader = new Utf8JsonReader(
+                    data.Span,
+                    isFinalBlock,
+                    state);
+            }
+            else if (segmentCount == 1)
+            {
+                reader = new Utf8JsonReader(
+                    new ReadOnlySequence<byte>(data),
+                    isFinalBlock,
+                    state);
+            }
+            else
+            {
+                reader = new Utf8JsonReader(
+                    JsonTestHelper.SegmentInto(data, segmentCount),
```

Comment:
What happens if data is too small for the given `segmentCount`?

For example, "true" with segment count 125?

Reply:
> What happens if data is too small for the given segmentCount?

It adds a bunch of empty segments.